<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MassPropertySetter</class>
 <widget class="QWidget" name="MassPropertySetter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>295</width>
    <height>128</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton{
	padding: 4 4 4 4px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="1">
      <widget class="Vec3Setter" name="widgetCOM" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>20</height>
        </size>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Inertia Tensor:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Mass(kg):</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Center of Mass:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QDoubleSpinBox" name="spinBoxMass">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="Matrix33Setter" name="widgetInertia" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>72</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="buttonSimplify">
           <property name="toolTip">
            <string>Computes a &quot;best&quot; approximation of a full inertia tensor as an oriented particle inertia tensor, which is constant along the diagonal, and zero on the off-diagonals, i.e., = I * k for some k. The matrix passed in is set to this approximation.</string>
           </property>
           <property name="text">
            <string>Simplify</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="buttonAutoFill">
           <property name="text">
            <string>Auto Fill</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Vec3Setter</class>
   <extends>QWidget</extends>
   <header>vec3setter.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Matrix33Setter</class>
   <extends>QWidget</extends>
   <header>matrix33setter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonSimplify</sender>
   <signal>clicked()</signal>
   <receiver>MassPropertySetter</receiver>
   <slot>onButtonSimplify()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonAutoFill</sender>
   <signal>clicked()</signal>
   <receiver>MassPropertySetter</receiver>
   <slot>onButtonAutoFill()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>buttonAutoFill(float)</signal>
  <slot>onButtonSimplify()</slot>
  <slot>onButtonAutoFill()</slot>
 </slots>
</ui>
